subprojects/core/src/main/java/org/gradle/api/artifacts/repositories/FlatDirectoryArtifactRepository.java

```diff
@@ -65,6 +65,14 @@
     /**
      * Sets the directories where this repository will look for artifacts.
      *
+     * @param dirs the directories.
+     * @since 4.0
+     */
+    void setDirs(Set<File> dirs);
```

Comment:
No, can't be `Iterable` as it would clash with the existing `void setDirs(Iterable<?> dirs)` because of type erasure ; can't be `Collection` as Kotlin won't pick the get/setters as a property.